### PR TITLE
Remove "null by omitting value" from blueprints

### PR DESCRIPTION
**Pull request by mr0re1**
* Remove "null by omitting value" from blueprints;
* Improve error message for wrong YAML type of `Dict`;
* Add tests for YAML null-values.

### CODE REVIEW

File: community/examples/hpc-slurm-local-ssd.yaml [hunk 1]
```diff
@@ -42,17 +42,17 @@ deployment_groups:
...
+        disk_name: null
         disk_size_gb: 375
         disk_type: local-ssd
-        disk_labels:
+        disk_labels: {}
         auto_delete: true
         boot: false
       - device_name: test-disk-2
-        disk_name:
+        disk_name: {}
```
Comment (tpdownes) [inline community/examples/hpc-slurm-local-ssd.yaml:52]: This should be `null`. It's causing a problem in PR validation.

**Review by tpdownes — APPROVED** [review]
My understanding is that we can't deny "omitted value" at parsing time due to our examples using `project_id: ####` so much. So, you are removing other omitted values from blueprints and adding better "plumbing" to test the 3 different ways a value can be `nil` in Go.

2 minor things to fix.
